fog/view/server/src/bin/router.rs

```diff
@@ -40,28 +41,39 @@ fn main() {
         logger.clone(),
     );
 
-    // TODO: Remove and get from a config.
-    let mut fog_view_store_grpc_clients = HashMap::new();
+    let mut shards = Vec::new();
     let grpc_env = Arc::new(
         grpcio::EnvBuilder::new()
             .name_prefix("Main-RPC".to_string())
             .build(),
     );
-    for shard_uri in config.shard_uris.clone() {
+    for (i, shard_uri) in config.shard_uris.clone().into_iter().enumerate() {
         let fog_view_store_grpc_client = FogViewStoreApiClient::new(
             ChannelBuilder::default_channel_builder(grpc_env.clone())
                 .connect_to_uri(&shard_uri, &logger),
         );
-        fog_view_store_grpc_clients.insert(shard_uri, Arc::new(fog_view_store_grpc_client));
+
+        let sharding_strategy = config
+            .sharding_strategies
+            .get(i)
+            .unwrap_or_else(|| panic!("Couldn't find shard at index {}", i));
+        let Epoch(epoch_sharding_strategy) = sharding_strategy;
+        let block_range = epoch_sharding_strategy.get_block_range();
+        let shard = Shard::new(
+            shard_uri.clone(),
```

Comment:
I'm pretty sure this clone is unnecessary, since you're cloning the collection at line 50.

Reply:
Done.